### ISSUE TITLE
[WP 6.7] Zoom Out toggle appears/disappears when Patterns inserter activates Zoom Out for non-iframed editor 

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#66884]
From further discovery [this is the true cause of the Issue](https://github.com/WordPress/gutenberg/issues/66884#issuecomment-2466091010)

> If a block with apiVersion below 2 is registered, the editor is not an iframe.
> If the editor is not an iframe, the zoom out toggle button will not be displayed.
> However, when you click the Patterns tab, the editor works as an iframe, so the zoom out toggle button appears.

---------

When too many plugins are registering items in the toolbar, the zoom toggle disappears in 6.7-RC4:

https://github.com/user-attachments/assets/c54a27fc-ea1a-43f5-9813-4f5390289309

Steps to replicate:
1. Install and activate Jetpack and Yoast.
2. Open up Pages > Add new.
3. Notice you don't see the zoom out toggle.
4. Notice that if you open up the Inserter > Patterns, that it appears.

I'd expect that the zoom out toggle would still be visible and, no matter how wide I try to make the viewport, it never shows up. cc @getdave I couldn't find a comparable issue on the 6.7 board for zoom out. 

Props to @laurelfulford for first finding this!

**Comment by annezazu** [comment]
cc @fabiankaegy as a heads up.

**Comment by laurelfulford** [comment]
We ran into the above when testing different Newspack plugins with WordPress 6.7. I've run into a wall trying to figure out why it's happening, but here's a few more details: 

We're seeing this with plugins like Jetpack, Newspack Blocks, Newspack Campaigns, Newspack Newsletters, Events Calendar, and some others we built. If just one of these plugins are enabled on their own (in Jetpack's case, enabled and connected), the Zoom out toggle goes away.

All of these plugins register blocks; when I removed the block registration code from a couple, the issue goes away. 

We also use WooCommerce, Co-Authors Plus, and Yoast, which also register blocks but don't cause this issue. 

One difference is that the latter group all register and enqueue block assets using block.json. Best I can tell the ones that cause issues with the Zoom out don't -- some of the blocks have a block.json, but don't seem to use it to register the block.

When the Zoom out isn't showing, opening the Pattern inserter will zoom the page out and make the toggle appear, but it disappears again once the Pattern inserter is closed. 

Hopefully this isn't a red herring! Just let me know if you have any questions about the above.

**Comment by annezazu** [comment]
Thank you so much for all of the extra details. Added props above for your hard work in trying to track this down and first reporting it :)

**Comment by t-hamano** [comment]
This issue is the same as #66671.

The cause of this issue is as follows:

- If a block with apiVersion below 2 is registered, the editor is not an iframe.
- If the editor is not an iframe, the zoom out toggle button will not be displayed.
- However, when you click the Patterns tab, the editor works as an iframe, so the zoom out toggle button appears.

This issue has been fixed in Gutenberg trunk (#66789), but we need to consider how to address this issue in WP 6.7.

My suggested approach is to either:

1. Do not enable zoom out mode in the Patterns tab
2. Enable the zoom out toggle too, which means applying #66671 to WP 6.7 too.

The latter approach is probably the best, but this would also be considered an enhancement, so I'm not sure if it's acceptable for a point release.

cc @stokesman @draganescu

**Comment by laurelfulford** [comment]
Thanks for tying that together, @t-hamano! 🙌 I focused too much on open issues when trying to search for something similar.

**Comment by getdave** [comment]
> Do not enable zoom out mode in the Patterns tab

As that's a Core piece of the UX for Zoom Out interactions (it's designed for composing with Patterns) so I can't see that being viable. 

> Enable the zoom out toggle too, which means applying https://github.com/WordPress/gutenberg/issues/66671 to WP 6.7 too.

I think you're suggesting applying https://github.com/WordPress/gutenberg/pull/66789 to WP 6.7? 

This is technically a bugfix but with the final release being _today_, ~in order for this to land it would need to be deemed either:~

- ~a _severe_ bug~
- ~something that locks us into a particular API in a way that could cause backwards compatibility issues~

~This Issue is clearly far from ideal, but does it _critically_ compromise the Editor? We also need to weigh this against any potential instability that could be caused by unknown side effects of enabling Zoom Out in the non-iframed editor.~

**Update**: The 24hour code freeze for the release is in place so we cannot include this in 6.7 final release and it will necessarily need to be punted to 6.7.1.

@annezazu @ndiego What is your feeling on the scope of this? 

I know also that @ellatrix [raised concerns about Zoom Out in non-iframe mode](https://github.com/WordPress/gutenberg/issues/66671#issuecomment-2460183754).

**Comment by ndiego** [comment]
Yes, I agree this will need to be punted to 6.7.1.

**Comment by annezazu** [comment]
Agreed to punt to 6.7.1 and I'd argue that it should be included as it's less of an enhancement and more of a bug in that you can't toggle this option on/off.

**Comment by t-hamano** [comment]
> I think you're suggesting applying https://github.com/WordPress/gutenberg/pull/66789 to WP 6.7?

Yes, I was referring to that PR.

**Comment by cbravobernal** [comment]
I'm moving this to 6.7.2 as is not a critical bug.

**Comment by MaggieCabrera** [comment]
Should we avoid going into zoom out mode on the first place when we are in this situation?

**Comment by t-hamano** [comment]
My suggestions for WordPress 6.7.2 are to either:

- Don't enable zoom out mode when you go to the Inserter > Patterns tab. That means if the editor is not an iframe, the zoom out mode is not available in any way in the post editor.
- Expose the zoom out toggle button to non-iframe editors too. That means backporting [#66789](https://github.com/WordPress/gutenberg/pull/66789) to WP 6.7.2. A manual backport might be needed.

**Comment by getdave** [comment]
> Expose the zoom out toggle button to non-iframe editors too. That means backporting https://github.com/WordPress/gutenberg/pull/66789 to WP 6.7.2. A manual backport might be needed.

Do you know if there are downsides to this? Otherwise it would seem the best route.

**Comment by t-hamano** [comment]
@getdave I found that zoom out mode doesn't work in the post editor in Gutenberg trunk.

#67232 seems to be related and is caused by [`hasSectionRootClientId`](https://github.com/WordPress/gutenberg/blob/72417c3b89b6c51fae4b1bf550833738f89aefaf/packages/editor/src/components/header/index.js#L107) being `false`. Is this the expected behavior and will the zoom out mode remain disabled in the post editor going forward?

If so, I'm wondering what the best approach is for the post editor in WP 6.7.2.

**Comment by getdave** [comment]
@t-hamano Yes it's my understanding that is expected.

That said, we might be able to adjust so that we can restore Zoom Out in the Post Editor but we need to avoid a situation whereby that would falsely enable Zoom Out in the Site Editor when the `main` or `post-content` (depending on Editor mode) do not exist.

**Comment by draganescu** [comment]
Just adding some reasoning for why this is expected: if we can't find a good section root the zoom out experience is highly likely to be broken. So until there are better options we should only enable the zoomed out UX where it's a fact that it will work as expected.

**Comment by t-hamano** [comment]
I tried applying #66789 on the `wp/6.7` branch: https://github.com/WordPress/gutenberg/compare/wp/6.7...zoom-out-non-iframed-6.7?expand=1

Technically, the zoom out toggle works in the non-iframe editor, but it's not visually smooth:

https://github.com/user-attachments/assets/b5a4e5dd-1c7a-4f4f-9baa-4a8c1ad4d10f

Of course, this is a known issue that occurs when you access the Patterns tab. Is this acceptable in WP 6.7?

**Comment by getdave** [comment]
Thank you for testing this. Perhaps if we just backported https://github.com/WordPress/gutenberg/pull/67232 to WP 6.7 this becomes a non-Issue. I don't see a huge value in Zoom Out when there is no true section root.

What do you think @richtabor @draganescu @annezazu?

**Comment by annezazu** [comment]
I agree completely.

**Comment by aaronjorbin** [comment]
Moving this off the 6.7.2 board based on discussion in the PR.

**Comment by t-hamano** [comment]
We decided NOT to address this issue in a minor release (6.7.2), see #68229 for details.

**Comment by TheJeffr0** [comment]
I've run into this issue today and wonder what the best course of action is. Should I wait for it to be patched in an upcoming release of WordPress or is there something I can do now in a theme to use the Zoom Out feature?

**Comment by t-hamano** [comment]
@TheJeffr0 Do you want to have zoom-out mode in the non-iframe post editor? If so, as of WP 6.7 it's not available.
See this code: https://github.com/WordPress/gutenberg/blob/1446c6b709590e4822f0c47c030e7bc0d1731b23/packages/editor/src/components/header/index.js#L153-L155

**Comment by tjasagithub** [comment]
Hi guys!
I just started using WordPress (ver 6.8.2) and the zoom button is not there no matter what I do. Anyone knows what I should do?

**Comment by t-hamano** [comment]
Hi @tjasagithub, 

As I understand it, the zoom out function should be available when the following conditions are met:

- The active theme is a block-based theme
- "Show template" mode is enabled
- The template contains a root section with a `main` tag

<img width="998" height="418" alt="Image" src="https://github.com/user-attachments/assets/4be9cc10-2938-46ef-99d6-bcbd2b127197" />
